lib/galaxy/datatypes/mothur.py

```diff
@@ -142,7 +150,7 @@ def set_meta(self, dataset, overwrite=True, skip=1, max_data_lines=100000, **kwd
             comment_lines = 0
             ncols = 0
 
-            headers = get_headers(dataset.file_name, sep='\t', count=max_data_lines)
+            headers = get_headers(dataset.file_name, sep='\t', count=-1)
```

Comment:
for the group files at least yes  ..these files never get crazy big, but we need to detect all the groups because many of the mothur tools use this metadata to let you select which groups to perform your analysis on.

I'll go over the file once more to see for which datatypes this is really necessary and for which perhaps not so important :)